iceoryx2-cal/src/static_storage/file.rs

```diff
@@ -213,10 +214,20 @@ impl crate::named_concept::NamedConceptMgmt for Storage {
     fn list_cfg(config: &Configuration) -> Result<Vec<FileName>, NamedConceptListError> {
         let msg = "Unable to list all storages";
         let origin = "static_storage::File::list_cfg()";
-        let directory = fail!(from origin, when Directory::new(&config.path),
-            map DirectoryOpenError::InsufficientPermissions => NamedConceptListError::InsufficientPermissions,
-            unmatched NamedConceptListError::InternalError,
-            "{} due to a failure while reading the storage directory (\"{}\").", msg, config.path);
+        let directory = match Directory::new(&config.path) {
+            Ok(directory) => directory,
+            Err(DirectoryOpenError::InsufficientPermissions) => {
+                debug!(from origin, "{} due to insufficient permissions to read the storage directory.", msg);
+                return Err(NamedConceptListError::InsufficientPermissions);
+            }
+            Err(DirectoryOpenError::DoesNotExist) => {
+                return Ok(vec![]);
```

Comment:
nitpick: the macro is not necessary and a `Ok(Vec::new())` would also work

Reply:
@elBoberido are you sure, the other suggestion is more to type? Why would it be better?

Reply:
Because `()` is easier to reach on my keyboard than `[]` ;)